Tools/wasm/wasi.py

```diff
@@ -264,7 +257,7 @@ def configure_wasi_python(context, working_dir):
 @subdir(lambda context: CROSS_BUILD_DIR / context.host_triple)
 def make_wasi_python(context, working_dir):
     """Run `make` for the WASI/host build."""
-    call(["make", "--jobs", str(cpu_count()), "all"],
+    call(["make", "--jobs", str(cpu_count()), "commoninstall"],
```

Comment:
Doesn't this (attempt to) trigger an install of Python itself?

https://github.com/python/cpython/blob/fca552993da32044165223eec2297b6aaaac60ad/Makefile.pre.in#L2220-L2222

That doesn't seem quite right to be triggering here as we aren't doing an install. For instance, `commoninstall` includes `altbininstall` which is used to install the interpreter:

https://github.com/python/cpython/blob/fca552993da32044165223eec2297b6aaaac60ad/Makefile.pre.in#L2248-L2251

Are you doing this to get `sysconfig` copied to `lib/` since the install commands blindly copy what's in that directory?

Reply:
I think doing an install in CI is good (as opposed to running the source directory) because I think downstream users are likely to want to do an install themselves. The CI is the most authoritative reference on how things should work for the downstream so in absence of other more important factors it's nice to make it helpful.

Because I set `--prefix={working_dir}` in the call to configure, it's installing into the build directory. Which is a bit sloppy but it works.

But what's actually needed here is that `commoninstall` implies `libinstall` and `sharedinstall`.

`libinstall` copies `$CHECKOUT/Lib` to `$BUILDDIR/lib` and also copies the sysconfigdata into `lib` here:
https://github.com/python/cpython/blob/21c04e1a972bd1b6285e0ea41fa107d635bbe43a/Makefile.pre.in#L2620-L2621

The other target that we use is `sharedinstall` which makes a `lib-dynload` folder and prevents `getpath.py` from printing `'Could not find platform dependent libraries <exec_prefix>'`. This message is harmless, but it's nice to suppress it.

Reply:
> I think doing an install in CI is good

This is used outside of CI where implicit installation isn't warranted.

This is getting all rather messy for WASI, so I think it's best to just leave it out and keep this focused on Emscripten.